### PR TITLE
Custom feeds on bsky appview

### PR DESCRIPTION
**Pull request by devinivy**
Sets-up the feeds on the bsky appview with proxying through the PDS.  Currently the PDS auths for the feedgen and the appview forwards those credentials along.  Small bit of followup work will be to perform some caching on the PDS to avoid always doing a feed generator service did lookup with the appview.

### CODE REVIEW

File: packages/pds/src/app-view/proxied/index.ts [hunk 1]
```diff
@@ -471,6 +474,31 @@ export default function (server: Server, ctx: AppContext) {
...
+    auth: ctx.accessVerifier,
+    handler: async ({ auth, params }) => {
+      const requester = auth.credentials.did
+      // @TODO cache the feedgen did lookup
+      const { data: feed } = await agent.api.app.bsky.feed
+        .getFeedGenerator({ feed: params.feed }, await headers(requester))
+        .catch((err) => {
+          console.log(err)
+          throw err
+        })
```
Comment (devinivy) [inline packages/pds/src/app-view/proxied/index.ts:487]: Reminder to self to remove the catch and log here.

File: packages/pds/src/app-view/proxied/index.ts [hunk 2]
```diff
@@ -471,6 +474,31 @@ export default function (server: Server, ctx: AppContext) {
...
+      if (!feed.view.did) {
```
Comment (devinivy) [inline packages/pds/src/app-view/proxied/index.ts:488]: We might want to address this with a lexicon change to make `did` required.

File: packages/bsky/service/index.js [hunk 3]
```diff
@@ -47,15 +55,22 @@ const main = async () => {
         pathPrefix: cfg.imgUriEndpoint && new URL(cfg.imgUriEndpoint).pathname,
       })
     : undefined
+  const algos = env.feedPublisherDid ? makeAlgos(env.feedPublisherDid) : {}
```
Comment (dholms) [inline packages/bsky/service/index.js:58]: i think you're missing an import on `makeAlgos`

File: packages/pds/src/app-view/proxied/index.ts (hunk 2 above)
Comment (dholms) [reply packages/pds/src/app-view/proxied/index.ts:488]: +1 yup it shoulda been

**Review by dholms — APPROVED** [review]
looks good 💪

File: packages/bsky/service/index.js (hunk 3 above)
Comment (devinivy) [reply packages/bsky/service/index.js:58]: Ohp, good call!  Exported it to use here but forgot to use it 🙃

File: packages/pds/src/app-view/proxied/index.ts (hunk 2 above)
Comment (devinivy) [reply packages/pds/src/app-view/proxied/index.ts:486]: Logged this.

File: packages/bsky/src/api/app/bsky/unspecced.ts [hunk 4]
```diff
@@ -12,13 +12,13 @@ export default function (server: Server, ctx: AppContext) {
...
       const { limit, cursor } = params
-      const requester = auth.credentials.did
+      const viewer = auth.credentials.did
       const db = ctx.db.db
       const { ref } = db.dynamic
 
       const feedService = ctx.services.feed(ctx.db)
-      const labelService = ctx.services.label(ctx.db)
 
+      // @TODO sync-up with pds, apply mutes and blocks
```
[the remark this answers is not in the document]
Comment (devinivy) [reply packages/bsky/src/api/app/bsky/unspecced.ts:21]: Logged this.

**Pull request merged**